test/versioned/pino/pino.tap.js

```diff
@@ -0,0 +1,320 @@
+    t.end()
+  })
+
+  t.test('local_decorating', (t) => {
+    setupAgent({
+      application_logging: {
+        enabled: true,
+        local_decorating: { enabled: true },
+        forwarding: { enabled: false },
+        metrics: { enabled: false }
+      }
+    })
+    const message = 'pino decorating test'
+    helper.runInTransaction(agent, 'pino-test', async () => {
+      logger.info(message)
+      const line = await once(stream, 'data')
+      t.ok(line.message.includes('NR-LINKING'), 'should contain NR-LINKING metadata')
+      contextKeys.forEach((key) => {
+        t.notOk(line[key], `should not have ${key}`)
+      })
+      t.end()
+    })
+  })
+
+  t.test('forwarding', (t) => {
+    let config
+    t.autoend()
+    t.beforeEach(() => {
+      config = setupAgent({
+        application_logging: {
+          enabled: true,
+          local_decorating: { enabled: false },
+          forwarding: { enabled: true },
+          metrics: { enabled: false }
+        }
+      })
+    })
+
+    t.test('should have proper metadata outside of a transaction', async (t) => {
+      const message = 'pino unit test'
+      const level = 'info'
+      logger[level](message)
+      const line = await once(stream, 'data')
+      t.validateAnnotations({ line, message, level, config })
+      t.equal(agent.logs.getEvents().length, 1, 'should have 1 log in aggregator')
+      t.same(agent.logs.getEvents(), [line], 'log should be same in aggregator')
+      t.end()
+    })
+
+    t.test('should have proper error keys when error is present', async (t) => {
+      const err = new Error('This is a test')
+      const level = 'error'
+      logger[level](err)
+      const line = await once(stream, 'data')
+      t.validateAnnotations({ line, config, message: err.message, level })
+      t.equal(line['error.class'], 'Error', 'should have Error as error.class')
+      t.equal(line['error.message'], err.message, 'should have proper error.message')
+      t.equal(line['error.stack'], truncate(err.stack), 'should have proper error.stack')
+      t.notOk(line.err, 'should not have err key')
+      t.equal(agent.logs.getEvents().length, 1, 'should have 1 log in aggregator')
+      t.same(agent.logs.getEvents(), [line], 'log should be same in aggregator')
+      t.end()
+    })
+
+    t.test('should add proper trace info in transaction', (t) => {
+      helper.runInTransaction(agent, 'pino-test', async (tx) => {
+        const level = 'info'
+        const message = 'My debug test'
+        logger[level](message)
+        const meta = agent.getLinkingMetadata()
+        const line = await once(stream, 'data')
+        t.validateAnnotations({ line, config, message, level })
+        t.equal(line['trace.id'], meta['trace.id'])
+        t.equal(line['span.id'], meta['span.id'])
+        t.equal(
+          agent.logs.getEvents().length,
+          0,
+          'should have not have log in aggregator while transaction is active'
+        )
```

Comment:
I like this test, very thorough to check it will only be aggregated after the txn ends.

Reply:
yea it's our only true integration tests of this behavior.  I also added it in the winston PR but I think we already had that in the winston enricher so i just adapted it to work in agent